imagedephi/utils/tiff.py

```diff
@@ -1,5 +1,33 @@
+from __future__ import annotations
+
+from collections.abc import Generator
+from pathlib import Path
+from typing import TYPE_CHECKING
+
 import tifftools
 
+if TYPE_CHECKING:
+    from tifftools.tifftools import IFD
+
+
+def iter_ifds(
+    ifds: list[IFD],
+    tag_set=tifftools.constants.Tag,
+) -> Generator[IFD, None, None]:
+    for ifd in ifds:
+        for tag_id, entry in ifd["tags"].items():
+            tag: tifftools.TiffTag = tifftools.constants.get_or_create_tag(
+                tag_id,
+                tagSet=tag_set,
+                datatype=tifftools.Datatype[entry["datatype"]],
+            )
+            if tag.isIFD():
+                # entry['ifds'] contains a list of lists
+                # see tifftools.read_tiff
+                for sub_ifds in entry.get("ifds", []):
+                    yield from iter_ifds(sub_ifds, tag.get("tagset"))
+        yield ifd
```

Comment:
It looks like an IFD iterator would be useful outside of the `RedactionPlan` classes. I think refactoring could be done in a follow up PR to keep the scope of this one down.